src/NuGet.Core/NuGet.ProjectModel/LockFile/AssetsLogMessage.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using NuGet.Common;
+
+namespace NuGet.ProjectModel
+{
+    public class AssetsLogMessage : RestoreLogMessage
+    {
+
+        public AssetsLogMessage(LogLevel logLevel, NuGetLogCode errorCode,
+            string errorString, string targetGraph)
+            :base(logLevel, errorCode, errorString, targetGraph)
+        {
+        }
+
+        public AssetsLogMessage(LogLevel logLevel, NuGetLogCode errorCode, string errorString)
+            : this(logLevel, errorCode, errorString, string.Empty)
+        {
+
+        }
+
+        public AssetsLogMessage(LogLevel logLevel, string errorString)
+            : this(logLevel, LogLevel.Error == logLevel ? NuGetLogCode.NU1000 : NuGetLogCode.NU1500, errorString, string.Empty)
+        {
+        }
+
+        /// <summary>
+        /// Used to convert the Log Message object into a dictionary that can be written into the assets file.
+        /// The Dictionary is converted into a JObject by the LockFileFormat.
+        /// </summary>
+        /// <returns>Dictionary of the properties representing the Log Message.</returns>
+
+        public JObject ToJObject()
```

Comment:
This feels like it should be in the LockFile writer and not the object itself now that the fields are all exposed.  I would just move this code there and keep the message as a simple class for storing the property values